db/types/base.py

```diff
@@ -109,9 +109,36 @@ def get_qualified_name(name):
 
 
 def get_available_types(engine):
+    """
+    Returns a dict where the keys are database type names defined on the database associated with
+    provided Engine, and the values are their SQLAlchemy classes.
+    """
     return engine.dialect.ischema_names
 
 
+_known_vanilla_db_types = tuple(postgres_type for postgres_type in PostgresType)
+
+
+_known_custom_db_types = tuple(mathesar_custom_type for mathesar_custom_type in MathesarCustomType)
+
+
+# Known database types are those that are defined on our PostgresType and MathesarCustomType Enums.
+known_db_types = _known_vanilla_db_types + _known_custom_db_types
+
+
+def get_available_known_db_types(engine):
+    """
+    Returns database types that are both available on the database and known through our Enums
+    above.
+    """
```

Comment:
This doesn't check the DB, but rather gets the types which are both available in the engine mapping, as well as the Enums.  Does it need to ensure the type is installed on the DB in order to fulfill its purpose?   If not, the doc string should change.  Otherwise, we should add a way to check that.

Reply:
Available in this context means that it's installed on the DB. I made a mistake thinking that the engine mapping is a proxy for that.

Reply:
I was going by the docstring on `db/types/base.py::get_available_types`:

```
def get_available_types(engine):
    """
    Returns a dict where the keys are database type names defined on the database associated with
    provided Engine, and the values are their SQLAlchemy classes.
    """
    return engine.dialect.ischema_names
```

Reply:
The original docstring was also a mistake it seems.